packages/perspective-viewer-d3fc/src/js/charts/bar.js

```diff
@@ -9,25 +9,31 @@
 import * as fc from "d3fc";
 import * as crossAxis from "../axis/crossAxis";
 import * as mainAxis from "../axis/mainAxis";
-import {barSeries, barColours} from "../series/barSeries";
-import {groupAndStackData} from "../data/groupAndStackData";
-import {legend, filterData} from "../legend/legend";
+import { barSeries, barColours } from "../series/barSeries";
+import { groupAndStackData } from "../data/groupAndStackData";
+import { legend, filterData } from "../legend/legend";
+import { mainGrid, crossGrid, gridlinesAnnotation, withGridLines } from "../gridlines/gridlines";

```

Comment:
I'm wondering if you could pull all this logic out into a single component that you'd call like this:

````javascript
const series = withGridLines(
    fc.seriesSvgMulti().mapping(... etc ...)
).orient("horizontal")();
````

Note the extra function call at the end. So `withGridLines` creates a D3-style component (i.e. it returns a function) with an `orient` property. When you later call the function, depending on whether `orient` is horizontal or vertical, it can then internally setup `gridLines` with `mainGrid` and `crossGrid` in the right order.

Reply:
I am actually just looking at doing something like that.

E.g.

const series = withGridLines(
        fc
            .seriesSvgMulti()
            .mapping((data, index) => data[index])
            .series(
                data.map(() =>
                    barSeries(settings, colour)
                        .align("left")
                        .orient("vertical")
                )
            )
    )
        .xStyle(crossGrid)
        .yStyle(mainGrid);

Reply:
`const series = withGridLines(
        fc
            .seriesSvgMulti()
            .mapping((data, index) => data[index])
            .series(
                data.map(() =>
                    barSeries(settings, colour)
                        .align("left")
                        .orient("vertical")
                )
            )
    )
        .xStyle(crossGrid)
        .yStyle(mainGrid);`

Reply:
I like the orient idea better, so I will try and do that.

Reply:
👍 
If you're always going to have either `.xStyle(crossGrid).yStyle(mainGrid)` or `.xStyle(mainGrid).yStyle(crossGrid)`, I thought it might be nicer to not have to import `mainGrid` and `crossGrid` into all the charts.

Reply:
Linting is now correct.

withGridLines refactored to be more in line with d3fc.
E.g.
`withGridLines(series).orient("horizontal")`